src/compiler/checker.ts

```diff
@@ -4515,7 +4515,7 @@ namespace ts {
                 }
                 // Handle export default expressions
                 if (declaration.kind === SyntaxKind.ExportAssignment) {
-                    return links.type = checkExpression((<ExportAssignment>declaration).expression);
+                    return links.type = checkExpression((<ExportAssignment>declaration).expression, checkMode);

```

Comment:
It doesn't seem like we should be holding on to the type here if this is not a normal check mode.

Reply:
Will this ever be anything apart from normal?

Reply:
Inferential. The reason its even being threaded through here is to catch when it is set to inferential.